### PR TITLE
Toast feature

### PR DESCRIPTION
**Pull request by abhisheksahani-nova**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,18 +13,35 @@ import Mockman from "mockman-js";
 import { useEffect } from "react";
 import "./index.css";
 import { useTheme } from "./context/theme-context";
+import { Toast } from "./components/index";
+import { useToast } from "./context/toast-context";
 
 function App() {
   const { theme } = useTheme();
+  const { toastData, setToastData } = useToast();
+  console.log(toastData);
```
Comment (govindappaarun) [inline src/App.js]: suggestion: remove console log

File: src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -16,7 +18,13 @@ function Navbar() {
     localStorage.removeItem("token");
     localStorage.removeItem("username");
     localStorage.removeItem("email");
+    setToastData({
+      showToast: true,
+      toastType: "success",
+      toastMesaage: "Successfully logout done ",
```
Comment (govindappaarun) [inline src/components/Navbar/Navbar.jsx]: suggestion: rewrite the sentence correctly

File: src/components/Toast/toast.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.alert-custom-style {
+  width: 35vw;
+  min-width: fit-content;
```
Comment (govindappaarun) [inline src/components/Toast/toast.css:3]: suggestion: instead use display: inline-block as fit-content as poor support from all browsers

File: src/context/cart-context.js [hunk 4]
```diff
@@ -18,8 +20,18 @@ const CartProvider = ({ children }) => {
...
       setCartProducts(response.data.cart);
+      setToastData({
+        showToast: true,
+        toastType: "success",
+        toastMesaage: "Successfully added to the cart",
+      });
     } catch (error) {
       console.log(error);
+      setToastData({
+        showToast: true,
```
Comment (govindappaarun) [inline src/context/cart-context.js]: suggestion: if its true always, you can avoid setting and make it default to reduce lines and default toast type as well

File: src/context/toast-context.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { useState, createContext, useContext, useEffect } from "react";
+
+const ToastContext = createContext();
+
+const ToastProvider = ({ children }) => {
+  const [toastData, setToastData] = useState({
+    showToast: false,
+    toastType: "",
+    toastMesaage: "",
```
Comment (govindappaarun) [inline src/context/toast-context.js]: suggestion: on naming, ` toast ` gets repeated on object and property as well eg. ` toastData.toastType ` can be simplified to ` toast.type `

**Review by govindappaarun — APPROVED** [review]
Looks good to merge with those comments fixed

File: src/App.js [hunk 6]
```diff
@@ -13,18 +13,35 @@ import Mockman from "mockman-js";
...
 
   useEffect(() => {
     localStorage.removeItem("token");
     localStorage.removeItem("username");
     localStorage.removeItem("email");
   }, []);
```
Comment (BalajiHarinath) [inline src/App.js]: Suggestion: Can use localStorage.clear() instead of removing the objects individually.

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggested changes everything else looks good.

File: src/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (abhisheksahani-nova) [reply src/components/Navbar/Navbar.jsx]: done

File: src/App.js (hunk 1 above)
Comment (abhisheksahani-nova) [reply src/App.js]: done

File: src/App.js (hunk 6 above)
Comment (abhisheksahani-nova) [reply src/App.js]: Yaah , thanks for suggesting that, I will do that

**Pull request merged**